### PR TITLE
Add compression

### PR DESCRIPTION
**Pull request by jovandeginste**

### CODE REVIEW

File: lib/puppet/provider/netscaler_service/rest.rb [hunk 1]
```diff
@@ -91,6 +91,7 @@ def property_to_rest_mapping
       :tcp_buffering       => :tcpb,
       :traffic_domain_id   => :td,
       :use_source_ip       => :usip,
+			:use_compression     => :cmp,
```
Comment (tphoney) [inline lib/puppet/provider/netscaler_service/rest.rb]: looks like the alignment is off on this.

File: lib/puppet/type/netscaler_service.rb [hunk 2]
```diff
@@ -232,6 +232,10 @@
     truthy_property("Use the client's IP address as the source IP address when initiating a connection to the server. When creating a service, if you do not set this parameter, the service inherits the global Use Source IP setting (available in the enable ns mode and disable ns mode CLI commands, or in the System > Settings > Configure modes > Configure Modes dialog box). However, you can override this setting after you create the service.", 'YES', 'NO')
   end
 
+  newproperty(:use_compression, :parent => Puppet::Property::NetscalerTruthy) do
+		truthy_property("Enable compression for the service.", 'YES', 'NO')
```
Comment (tphoney) [inline lib/puppet/type/netscaler_service.rb]: alignment is off.

**Comment by tphoney** [comment]
Looks good, could you correct the alignment issues. Rubocop can fix it automatically. https://github.com/bbatsov/rubocop

**Comment by jovandeginste** [comment]
Should be re-aligned and squahed.

I couldn't get rubocop working (quickly)...

**Comment by tphoney** [comment]
Superb work :), i squashed your commits.

**Comment by jovandeginste** [comment]
They were not squashed?

**Comment by tphoney** [comment]
nope, i squashed them on merge. It was easy to do :+1:

**Comment by jovandeginste** [comment]
Yes but I thought I squashed them before last push ... Have to find out what went wrong then!

**Comment by jovandeginste** [comment]
I do see only one single commit here: https://github.com/puppetlabs/puppetlabs-netscaler/pull/186/commits

Care to explain what 'squashing' means? Maybe I'm confused :-)

**Comment by tphoney** [comment]
https://github.com/puppetlabs/puppetlabs-netscaler/commit/977daad148aa53fc56e3b8259abbd44ad349695f is the commit hash, i merged the commits into one. I took the 2 commit message into one. 

There is no issue. Try it again for #181

**Comment by tphoney** [comment]
https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Squashing-Commits

**Comment by jovandeginste** [comment]
Yes, that is what I did ... Anyway, I'll try again with the others :-)

I'm writing scripts to generate the type+provider from Citrix' documentation. Another tool to generate puppet manifest files from 'show run' output. Should I include those with this repo? If so, where?

**Pull request merged**